src/System.Private.CoreLib/src/System/Threading/Lock.cs

```diff
@@ -14,8 +14,13 @@ public sealed class Lock
         // class construction.  If Lock has its own class constructor, this can lead to infinite recursion.
         // All static data in Lock must be pre-initialized.
         //
-        [PreInitialized]
-        private static int s_maxSpinCount = -1; // -1 means the spin count has not yet beeen determined.
+        private static int s_maxSpinCount;
+
+        // Eager construction called from LibraryInitialize
+        internal static void Initialize()
+        {
+            s_maxSpinCount = -1; // -1 means the spin count has not yet beeen determined.
```

Comment:
It would look nicer to changes this to mean: 0 - not initialized; -1 - no spinning; >0 - actual spin count. It would avoid the special plumbing for initialing this to -1.